pkgs/development/python-modules/black/newer-setuptools_scm.patch

```diff
@@ -0,0 +1,12 @@
+Upstream patch with whitespace change:
+https://github.com/psf/black/commit/41e670064063e3e221b1c3c40db5aaae784b5231.patch
+--- a/pyproject.toml
++++ b/pyproject.toml
+@@ -23,1 +23,1 @@ extend-exclude = '''
+-requires = ["setuptools>=41.0", "setuptools_scm~=6.0.1", "wheel"]
++requires = ["setuptools>=45.0", "setuptools_scm[toml]>=6.3.1", "wheel"]
```

Comment:
this should probably be
```suggestion
+requires = ["setuptools>=45.0", "setuptools_scm[toml]~=6.0", "wheel"]
```

assuming that setuptools_scm does do breaking changes between major versions

Also, I would prefer the usage of `substituteInPlace` as editing patches is more of a maintenance burden

Reply:
To shed some light on my perspective: all I see is a build failure in attempt to build `nixos/system` against local system's config. Looks like `python-black` is the only failing package. I don't really know anything about python packaging or python ecosystem.

The change was picked from upstream at https://github.com/psf/black/commit/41e670064063e3e221b1c3c40db5aaae784b5231 as is. It seems to survive build and test against nixpkgs' `python3.9-setuptools-scm-6.3.1`. Is your worry that it's an incorrect or incomplete change?

WRT maintainability the change should be dropped entirely on next release of `black`. Be it `substituteInPlace` or a standalone patch. I had an impression that `substituteInPlace` does not fail if it stops doing intended change. If it's an intended use I can change it to `substituteInPlace`.

Pinning the package back to `setuptools-scm-6.0` would require pulling in older version into `nixpkgs`. It should be doable, but then it would probably not require any package change. Should I try it?